tests/unit/synapseclient/unit_test_tables.py

```diff
@@ -119,6 +121,106 @@ def test_cast_values__list_type():
     ]
 
 
+def test_convert_df_date_cols_to_datetime_empty_df():
+    test_df = pd.DataFrame()
+    date_columns = []
+    test_df_output = _convert_df_date_cols_to_datetime(test_df, date_columns)
+    assert test_df_output.empty
+
+
+def test_convert_df_date_cols_to_datetime_invalid_epoch_time():
+    test_df = pd.DataFrame(
+        {
+            "epoch_time1": [
+                "1107234000000",
+                "invalid epoch time",
+                "invalid epoch time",
+                "invalid epoch time",
+                "invalid epoch time",
+            ]
+        }
+    )
+    date_columns = ["epoch_time1"]
+    with pytest.raises(ValueError):
+        test_df_output = _convert_df_date_cols_to_datetime(test_df, date_columns)
+
+
+def test_convert_df_date_cols_to_datetime_invalid_date_columns():
+    test_df = pd.DataFrame(
+        {
+            "epoch_time1": [
+                "1107234000000",
+                "1107320400000",
+                "1107406800000",
+                "1107493200000",
+                "1107579600000",
+            ]
+        }
+    )
+    date_columns = ["invalid date column"]
+    with pytest.raises(ValueError):
+        test_df_output = _convert_df_date_cols_to_datetime(test_df, date_columns)
+
+
+def test_convert_df_date_cols_to_datetime():
+    # construct test dataframe and get date columns
+    test_df = pd.DataFrame(
+        {
+            "row": [1, 2, 3, 4, 5],
+            "epoch_time1": [
+                "1107234000000",
+                "1107320400000",
+                "1107406800000",
```

Comment:
Add in a `float('nan')` and see what happens.

Reply:
Okay. I have added float('nan') in my test